internal/command/machine/clone.go

```diff
@@ -103,8 +103,25 @@ func runMachineClone(ctx context.Context) (err error) {
 	}
 	flapsClient := flaps.FromContext(ctx)
 
+	var vol *api.Volume
+	if volumeInfo := flag.GetString(ctx, "attach-volume"); volumeInfo != "" {
```

Comment:
A downside of this implementation is that the logic is duplicated further down (see below). I thought about storing the results in variables and using them further below but the logic started to get kind of messy. Still trying to see if I can get it right but if this is fine as is, happy to leave it.

https://github.com/superfly/flyctl/blob/7c5e9480d437495e37a3f1c572ec2b84614cdae9/internal/command/machine/clone.go#L177-L214

Reply:
duplicated code is definitely gross but I think it's probably fine here. plumbing everything through would probably end up reading less clearly.